eventnotifier/http/http.go

```diff
@@ -46,28 +48,48 @@ type Notifier struct {
 	NotificationTitle  string                 `jsonapi:"attribute" json:"notification_title"`
 	NotificationType   types.NotificationType `jsonapi:"attribute" json:"notification_type"`
 	EventMessage       string                 `jsonapi:"attribute" json:"event_message"`
 	InvolvedObjectKind string                 `jsonapi:"attribute" json:"involved_object_kind"`
-	DisruptionName     string                 `jsonapi:"attribute" json:"disruption_name"`
-	Disruption         string                 `jsonapi:"attribute" json:"disruption"`
-	Timestamp          int64                  `jsonapi:"attribute" json:"timestamp"`
-	Cluster            string                 `jsonapi:"attribute" json:"cluster"`
-	Namespace          string                 `jsonapi:"attribute" json:"namespace"`
-	TargetsCount       int                    `jsonapi:"attribute" json:"targets_count"`
-	Username           string                 `jsonapi:"attribute" json:"username,omitempty"`
-	UserEmail          string                 `jsonapi:"attribute" json:"user_email,omitempty"`
-	UserGroups         string                 `jsonapi:"attribute" json:"user_groups,omitempty"`
+	// Deprecated: DisruptionName exists for historical compatibility
+	// and should not be used. Use DisruptionEvent.DisruptionName instead.
+	DisruptionName string `jsonapi:"attribute" json:"disruption_name"`
+	// Deprecated: Disruption exists for historical compatibility
+	// and should not be used. Use DisruptionEvent.Disruption instead.
+	Disruption string `jsonapi:"attribute" json:"disruption"`
+	// Deprecated: TargetsCount exists for historical compatibility
+	// and should not be used. Use DisruptionEvent.TargetsCount instead.
+	TargetsCount        int                  `jsonapi:"attribute" json:"targets_count"`
+	Timestamp           int64                `jsonapi:"attribute" json:"timestamp"`
+	Cluster             string               `jsonapi:"attribute" json:"cluster"`
+	Namespace           string               `jsonapi:"attribute" json:"namespace"`
+	Username            string               `jsonapi:"attribute" json:"username,omitempty"`
+	UserEmail           string               `jsonapi:"attribute" json:"user_email,omitempty"`
+	UserGroups          string               `jsonapi:"attribute" json:"user_groups,omitempty"`
+	DisruptionEvent     *DisruptionEvent     `jsonapi:"attribute" json:"disruption_event,omitempty"`
+	DisruptionCronEvent *DisruptionCronEvent `jsonapi:"attribute" json:"disruption_cron_event,omitempty"`
+}
+
+type DisruptionEvent struct {
+	DisruptionName string `jsonapi:"attribute" json:"disruption_name"`
+	Disruption     string `jsonapi:"attribute" json:"disruption"`
+	TargetsCount   int    `jsonapi:"attribute" json:"targets_count"`
+}
+
+type DisruptionCronEvent struct {
+	DisruptionCronName string `jsonapi:"attribute" json:"disruption_cron_name"`
+	DisruptionCron     string `jsonapi:"attribute" json:"disruption_cron"`
```

Comment:
Could we make it clearer that the Disruption and DisruptionCron fields should be the full manifests?

Reply:
```suggestion
type DisruptionCronEvent struct {
	NotifierEvent
	Name     string `jsonapi:"attribute" json:"name"`
	Manifest string `jsonapi:"attribute" json:"manifest"`
}
```
What do you think of this struct?

Reply:
```
type DisruptionCronEvent struct {
	NotifierEvent
	Name     string `jsonapi:"attribute" json:"name"`
	Manifest string `jsonapi:"attribute" json:"manifest"`
}

type DisruptionEvent struct {
	NotifierEvent
	Name         string `jsonapi:"attribute" json:"name"`
	Manifest     string `jsonapi:"attribute" json:"manifest"`
	TargetsCount int    `jsonapi:"attribute" json:"targets_count"`
}

```

What do you think of those structs?

Reply:
I certainly think it's more clear. And we're breaking the api anyway with the restructuring, so I'm +1. @diyarab and @clairecng ?